/*
    Copyright 2018 Flight Lab authors.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

       https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto3";

import "google/protobuf/empty.proto";

package flightlab;

// Service provided by client machine.
service ClientService {
  // Displays a text message.
  rpc DisplayMessage(Message) returns (GeneralResponse) {
  }
  // Displays an image.
  rpc DisplayImage(Image) returns (GeneralResponse) {
  }
  // Turns off display.
  rpc DisplayOff(google.protobuf.Empty) returns (GeneralResponse) {
  }
}

message Message {
  string message = 1;
}

message Image {
  string image_path = 2;
}

message GeneralResponse {
  bool succeed = 1;
  string error_message = 2;
}
